kubeflow/gcp/basic-auth-ingress.libsonnet

```diff
@@ -430,6 +436,45 @@
     ),
     cloudEndpoint:: cloudEndpoint,
 
+    local istioMappingSvc = {
+      apiVersion: "v1",
+      kind: "Service",
+      metadata: {
+        labels: {
+          app: "istioMappingSvc",
+        },
+        name: "istio-mapping-service",
+        namespace: namespace,
+        annotations: {
+          "getambassador.io/config":
+            std.join("\n", [
+              "---",
+              "apiVersion: ambassador/v0",
+              "kind:  Mapping",
+              "name: istio-mapping",
+              "prefix_regex: true",
+              "prefix: /(?!whoami|kflogin).*",
+              "rewrite: \"\"",
```

Comment:
this is "" ?
Why should rewrite be this?

Reply:
rewrite being this so that it's not rewritten.  this is a hack to redirect all traffic to istio ingress gateway and the actual rewrite will be there, according to virtual service configs.

Reply:
If we don't want rewrite, can we remove this line?
Or it won't work?

Reply:
If we don't put this line, default will rewrite prefix to / and it will be wrong when passing to ISTIO.